src/routes/ui.rs

```diff
@@ -1,31 +1,53 @@
 use rocket::Route;
-use rocket_contrib::Template;
+use maud::{html, Markup, PreEscaped};
 
 use db;
 use db::models::Account;
+use templates::{Page, PageBuilder};
 use error::Perhaps;
 
 pub fn routes() -> Vec<Route> {
     routes![index, user_page]
 }
 
 #[get("/users/<username>", format = "text/html")]
-pub fn user_page(username: String, db_conn: db::Connection) -> Perhaps<Template> {
+pub fn user_page(username: String, db_conn: db::Connection) -> Perhaps<Page> {
     let account = try_resopt!(Account::fetch_local_by_username(&db_conn, username));
 
-    // We can use a cute hack to remove the need to explicitly write out a context struct,
-    // by using the serde_json helper to construct a `Serialize`-able struct on the fly.
-    let context = json!({
-        "fq_username": account.fully_qualified_username(),
-        "display_name": account.display_name.as_ref().unwrap_or(&account.username),
-        "uri": &account.get_uri(),
-        "bio": account.summary.as_ref().map(String::as_str).unwrap_or("<p></p>"),
-    });
+    let rendered = PageBuilder::default()
+        .title(format!("@{user}", user = account.username))
+        .content(html! {
+            div.h-card {
+                header {
+                    h1 a.u-url.u-uid href=(account.get_uri()) {
+                        span.p-name (account.display_name.as_ref().unwrap_or(&account.username))
+                    }
 
-    Ok(Some(Template::render("user_profile", context)))
+                    div (account.fully_qualified_username())
+                }
+
+                div.p-note {
+                    @if let Some(bio) = account.summary.as_ref() {
+                        (PreEscaped(bio))
+                    } @else {
+                        p {}
+                    }
+                }
+            }
+        })
+        .build()
+        .unwrap();
```

Comment:
Is unwrap safe here?

Reply:
Yes, since we satisfy `derive_builder`'s `build()` constraints: namely, that `content` must be provided.

Reply:
If there's a nicer way to do this, I'd do it, but I don't think there is?

Reply:
Eh. Probably not. I just don't like seeing unwraps, it makes me fear the possibility of a panic.

I'm okay with this, it might be useful adding a comment explaining why the unwrap is definitely safe.

Reply:
Oh, fwiw, Rocket catches panics, unwinds them, and delivers a 500.

Reply:
Yeah, I think I'll do that. Also, I should note that Rocket catches panics, unwinds them, and 500s.